clang/test/SemaSYCL/lb_sm_70.cpp

```diff
@@ -0,0 +1,60 @@
+// RUN: %clang_cc1 -internal-isystem %S/Inputs -triple nvptx-unknown-unknown -target-cpu sm_70 -fsycl-is-device -Wno-c++23-extensions -S -emit-llvm %s -o -ferror-limit=100 -fsyntax-only -verify %s
+
+// Maximum work groups per multi-processor, mapped to maxclusterrank PTX
+// directive, is an SM_90 feature, make sure that correct warning is issued on
+// architectures lower than that. Furthermore, warn/error incorrect values
+// specified for max_work_groups_per_mp and min_work_groups_per_cu.
+
+#include "sycl.hpp"
+
+template <int N1, int N2, int N3> class Functor {
+public:
+  // expected-warning@+2 {{'maxclusterrank' requires sm_90 or higher, CUDA arch provided: sm_70, ignoring 'max_work_groups_per_mp' attribute}}
+  [[intel::max_work_group_size(1, 1, N1), intel::min_work_groups_per_cu(N2),
+    intel::max_work_groups_per_mp(N3)]] void
+  operator()() const {}
+};
+
+int main() {
+  sycl::queue Q{};
+
+  Q.submit([&](sycl::handler &cgh) {
+    // expected-warning@+4 {{'maxclusterrank' requires sm_90 or higher, CUDA arch provided: sm_70, ignoring 'max_work_groups_per_mp' attribute}}
+    cgh.single_task<class T1>(
+        [=] [[intel::max_work_group_size(1, 1, 256),
```

Comment:
What should happen if a user inccorectly specifies `intel::min_work_groups_per_cu(2)` or `intel::max_work_groups_per_mp(4)]` without `[[intel::max_work_group_size(1, 1, 256)`?

Reply:
For now it's just silently dropped. Do you think there should be a warning that:
* `max_work_groups_per_mp` requires `max_work_group_size` and `min_work_groups_per_cu`
* `min_work_groups_per_cu` requires `max_work_group_size`

Reply:
This review has already grown arms and legs, I've got a follow up PR addressing the issue here: https://github.com/jchlanda/llvm/pull/1
I'll change target branches and submit it against intel/llvm once this PR goes in.